frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -0,0 +1,273 @@
+};
+
+export function createParameter(
+  option: ParameterMappingOptions,
+  parameters: Parameter[] = [],
+): Parameter {
+  let name = option.combinedName || option.name;
+  let nameIndex = 0;
+  // get a unique name
+  while (_.any(parameters, p => p.name === name)) {
+    name = (option.combinedName || option.name) + " " + ++nameIndex;
+  }
+
+  const parameter = {
+    name: "",
+    slug: "",
+    id: Math.floor(Math.random() * Math.pow(2, 32)).toString(16),
+    type: option.type,
+    sectionId: option.sectionId,
+  };
+  return setParameterName(parameter, name);
+}
+
+export function setParameterName(
+  parameter: Parameter,
+  name: string,
+): Parameter {
+  if (!name) {
+    name = "unnamed";
+  }
+  const slug = slugify(name);
+  return {
+    ...parameter,
+    name: name,
+    slug: slug,
+  };
+}
+
+export function setParameterDefaultValue(
+  parameter: Parameter,
+  value: any,
+): Parameter {
+  return {
+    ...parameter,
+    default: value,
+  };
+}
+
+export function hasMapping(parameter: Parameter, dashboard: Dashboard) {
+  return dashboard.ordered_cards.some(ordered_card => {
+    return ordered_card?.parameter_mappings?.some(parameter_mapping => {
+      return parameter_mapping.parameter_id === parameter.id;
+    });
+  });
+}
+
+export function isDashboardParameterWithoutMapping(
+  parameter: Parameter,
+  dashboard: Dashboard,
+) {
+  if (!dashboard) {
+    return false;
+  }
+
+  const parameterExistsOnDashboard = dashboard.parameters.some(
+    dashParam => dashParam.id === parameter.id,
+  );
+  const parameterHasMapping = hasMapping(parameter, dashboard);
+
+  return parameterExistsOnDashboard && !parameterHasMapping;
+}
+
+export function getDashboardUiParameters(
+  dashboard: Dashboard,
+  metadata: Metadata,
+): UiParameter[] {
+  const { parameters, ordered_cards } = dashboard;
+  const mappings = getMappings(ordered_cards);
+  const uiParameters: UiParameter[] = parameters.map(parameter => {
+    if (isFieldFilterParameter(parameter)) {
+      return buildFieldFilterUiParameter(parameter, mappings, metadata);
+    }
+
+    return {
+      ...parameter,
+    };
+  });
+
+  return uiParameters;
+}
+
+function buildFieldFilterUiParameter(
+  parameter: Parameter,
+  mappings: Mapping[],
+  metadata: Metadata,
+): FieldFilterUiParameter {
+  const mappingsForParameter = mappings.filter(
+    mapping => mapping.parameter_id === parameter.id,
+  );
+  const mappedFields = mappingsForParameter.map(mapping => {
+    const { target, card } = mapping;
+    return getTargetField(target, card, metadata);
+  });
+  const fields = mappedFields.filter((field): field is Field => field != null);
+  const hasOnlyFieldTargets =
+    mappedFields.length !== 0 && mappedFields.length === fields.length;
+  const uniqueFieldsWithFKResolved = _.uniq(
+    fields.map(field => field.target ?? field),
+    field => field.id,
+  );
+
+  return {
+    ...parameter,
+    fields: uniqueFieldsWithFKResolved,
+    hasOnlyFieldTargets,
```

Comment:
Previously, we were adding `field_ids` and `field_id` to this parameters object. With our recent refactor of `ParameterValueWidget`, we can actually totally remove them, with a minor tweak to `ParameterSidebar` to use `fields` instead of `field_ids`.